server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -414,6 +422,60 @@ private void createMetadataBasedIterables(Metadata metadata) {
         routines = () -> sequentialStream(routineInfos).filter(Objects::nonNull).iterator();
     }
 
+    public Set<String> enabledRoles(Role role, Roles roles) {
+        Set<String> result = new HashSet<>();
+        Set<ApplicableRole> applicableRoles = applicableRoles(role, roles);
+        for (ApplicableRole applicableRole: applicableRoles) {
+            result.add(applicableRole.roleName());
+        }
+        return result;
+    }
+
+    public Set<ApplicableRole> applicableRoles(Role role, Roles roles) {
+        Set<ApplicableRole> applicableRoles = new HashSet<>();
+
+        for (GrantedRole grantedRole: role.grantedRoles()) {
+            ApplicableRole newRole = new ApplicableRole(
+                role.name(),
+                grantedRole.roleName(),
+                roles.hasPrivilege(role, Permission.AL, Securable.CLUSTER, null)
+            );
+            Role retrievedRole = roles.findRole(grantedRole.roleName());
+            if (retrievedRole != null) {
+                if (!retrievedRole.grantedRoles().isEmpty()) {
+                    applicableRoles.addAll(applicableRoles(retrievedRole, roles));
+                }
+            }
+            applicableRoles.add(newRole);
+        }
+
+        return applicableRoles;
+    }
+
+    public Iterable<RoleTableGrant> roleTableGrants(Role role, Roles roles) {
+        List<RoleTableGrant> roleTableGrants = new ArrayList<>();
+        role.privileges().forEach(p -> {
+            Securable securableType = p.subject().securable();
+            if (securableType.equals(Securable.TABLE)) {
+                String[] fqn = Objects.requireNonNull(p.subject().ident()).split("\\.");
+
+                RoleTableGrant roleTableGrant = new RoleTableGrant(
+                    p.grantor(),
+                    role.name(),
+                    "doc",
+                    fqn[0],
+                    fqn[1],
+                    p.subject().permission().name(),
+                    roles.hasPrivilege(role, Permission.AL, Securable.CLUSTER, null),
+                    p.subject().permission().equals(Permission.DQL)
+                );
+                roleTableGrants.add(roleTableGrant);
+            }
+        });
+
+        return roleTableGrants;
```

Comment:
Same here, this could be replaced by a 
`sequentialStream(role.privileges()).filter(p -> p.subject().securable().equals(Securable.TABLE)).map(...)` function